keras/layers/preprocessing/discretization_test.py

```diff
@@ -125,6 +126,15 @@ def test_saving(self):
         model = saving_api.load_model(fpath)
         self.assertAllClose(layer(ref_input), ref_output)
 
+    @pytest.mark.skipif(
+        backend.backend() != "tensorflow",
+        reason="Sparse tensor only works in TensorFlow",
+    )
     def test_sparse_inputs(self):
-        # TODO
-        pass
+        from keras.utils.module_utils import tensorflow as tf
+
+        x = tf.sparse.from_dense(np.array([[-1.0, 0.2, 0.7, 1.2]]))
+        layer = layers.Discretization(bin_boundaries=[0.0, 0.5, 1.0])
```

Comment:
I think the idea would be to test with `sparse=True` with modes "one_hot", "multi_hot", and "count", and check that the output is a sparse tensor. This should be the case whether the input is sparse or not.

Reply:
@fchollet 👌

Reply:
@fchollet Is it test_spare_output ? I am getting confused.
Since we can convert a SparseTensor into a DenseTensor using tf.sparse.to_dense